miniconf_derive/src/lib.rs

```diff
@@ -124,25 +119,18 @@ fn metadata_arm((i, struct_field): (usize, &StructField)) -> proc_macro2::TokenS
         quote! {
             #i => {
                 let mut meta = <#field_type>::metadata();
-
-                // Unconditionally account for separator since we add it
-                // even if elements that are deferred to (`Options`)
-                // may have no further hierarchy to add and remove the separator again.
```

Comment:
This doc is important and seems to me to be still acurate. Why'd we remove it?

Reply:
It's wrong. We simply always add it. And that's now a redundant statement as any operation on the path anywhere starts with adding the separator. And the statement about removing the separator is wrong as we never need to remove the separator anymore.